edgedb-client/src/builder.rs

```diff
@@ -322,6 +329,171 @@ impl fmt::Display for DisplayAddr<'_> {
     }
 }
 
+impl DSN {
+    fn new(dsn: &str) -> Result<Self, Error> {
```

Comment:
This could be `FromStr for Dsn`. But since the type is not public, could be just `from_str`.